vendor/k8s.io/kube-openapi/pkg/schemaconv/smd.go

```diff
@@ -312,6 +312,18 @@ func (c *convert) VisitKind(k *proto.Kind) {
 			NamedType: &deducedName,
 		}
 	}
+
+	ext := k.GetExtensions()
+	if val, ok := ext["x-kubernetes-map-type"]; ok {
+		switch val {
+		case "atomic":
+			a.Map.ElementRelationship = schema.Atomic
+		case "separable":
+			a.Map.ElementRelationship = schema.Separable
+		default:
+			c.reportError("unknown map type %v", val)
+		}
+	}
```

Comment:
also, should granular be the default, rather than unset?

Reply:
The convention is to leave the value unset for SMD schemas if it is granular to keep the YAML representation from being too verbose.

Reply:
oh... I didn't know this affected serialization... does leaving ElementRelationship unset have the same behavior as setting it to granular?

Reply:
Yeah, unset is the same behavior as granular.